app/vmauth/main.go

```diff
@@ -345,12 +346,7 @@ var (
 	missingRouteRequests     = metrics.NewCounter(`vmauth_http_request_errors_total{reason="missing_route"}`)
 )
 
-var (
-	transport     *http.Transport
-	transportOnce sync.Once
-)
-
-func transportInit() {
+func getTransport(skipTLSVerification bool) *http.Transport {
```

Comment:
This change increases resource usage and slows vmauth performance. Possible workaround:

 create an additional instance with disable tls and select transport based on enabled/disabled tls.

Reply:
Added a simple cache to reuse already created transport with the same TLS verification option.

Reply:
I don't really like sync.Map for this case. 

 As an option. We can create http.Transport per auth config record. It'll simplify code for enterprise version with full tls options support.